src/Logic.hs

```diff
@@ -369,16 +369,29 @@ handlePullRequestEdited prId newTitle newBaseBranch state =
     -- Do nothing if the pull request is not present.
     Nothing -> pure state
 
+data ParseResult a
+  = Success a
+  | Unknown Text
+  | Ignored
+
+isSuccess :: ParseResult a -> Bool
+isSuccess (Success _) = True
+isSuccess _ = False
+
 -- Returns the approval type contained in the given text, if the message is a
 -- command that instructs us to merge the PR.
 -- If the trigger prefix is "@hoffbot", a command "@hoffbot merge" would
 -- indicate the `Merge` approval type.
-parseMergeCommand :: TriggerConfiguration -> Text -> Maybe (ApprovedFor, MergeWindow)
+-- Returns `Ignored` if the bot was not mentioned, `Success` if it was mentioned
+-- in the same message as a valid command, and `Unknown` if it was mentioned but
+-- no valid command was found.
+parseMergeCommand :: TriggerConfiguration -> Text -> ParseResult (ApprovedFor, MergeWindow)
 parseMergeCommand config message =
   let
     messageCaseFold = Text.toCaseFold $ Text.strip message
     prefixCaseFold = Text.toCaseFold $ Config.commentPrefix config
     infixMatch msg = (prefixCaseFold <> msg) `Text.isInfixOf` messageCaseFold
+    mentioned = prefixCaseFold `Text.isPrefixOf` message
```

Comment:
We should also case fold the message, otherwise the `isPrefixOf` will not work with incorrect cases.